src/components/common/CheckboxField.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { useFormContext } from 'react-hook-form';
+
+interface CheckboxFieldProps {
+  name: string;
+  label: string;
+  required?: boolean;
+}
+
+const CheckboxField: React.FC<CheckboxFieldProps> = ({
+  name,
+  label,
+  required = false
+}) => {
```

Comment:
isn't the syntax normally that the props come after the argument? Can you please double check this?